lib/OpenQA/Worker/Job.pm

```diff
@@ -439,9 +439,15 @@ sub _stop_step_5_upload {
         }
 
         # upload other logs and files
-        for my $file (
-            qw(video.ogv video_time.vtt vars.json serial0 autoinst-log.txt serial_terminal.txt virtio_console.log worker-log.txt)
-          )
+        my @files_to_upload;
+        if (defined $self->info->{settings}->{AUTOINST_UPLOAD_FROM_POOL}) {
+            @files_to_upload = split(',', $self->info->{settings}->{AUTOINST_UPLOAD_FROM_POOL});
+        }
+        else {
+            log_error('AUTOINST_UPLOAD_FROM_POOL is not defined fallback to default list');
+            @files_to_upload = qw(video.ogv video_time.vtt vars.json serial0 autoinst-log.txt serial_terminal.txt virtio_console.log worker-log.txt);
+        }
```

Comment:
Replace the whole block by

```perl
use constant UPLOAD => qw(video.ogv video_time.vtt vars.json serial0 autoinst-log.txt serial_terminal.txt virtio_console.log worker-log.txt);
…
my @files_to_upload = split(',', $self->info->{settings}->{AUTOINST_UPLOAD_FROM_POOL} // '') // UPLOAD;
```

Reply:
I'm not sure if it produce the output what you expect. Look at this:
```
perl -e "use Data::Dumper; @a = qw/foo bla/; @b = split(',','') // @a; print Dumper(\@b);"
```

Reply:
just use `||` then